Elektra with BUILD_TESTING=OFF fails

## Steps to Reproduce the Problem

Build Elektra with:
```
-DBUILD_TESTING=OFF
-DPLUGINS="ALL;-multifile;-simpleini"
-DBINDINGS="MAINTAINED;-intercept_env;-intercept_fs"
```

## Expected Result

Elektra compiles.

## Actual Result

```
`CMake Error at src/bindings/io/uv/CMakeLists.txt:61 (add_executable):
  Error evaluating generator expression:

    $<TARGET_OBJECTS:cframework>

  Objects of target "cframework" referenced but no such target exists.


CMake Error at src/bindings/io/glib/CMakeLists.txt:52 (add_executable):
  Error evaluating generator expression:

    $<TARGET_OBJECTS:cframework>

  Objects of target "cframework" referenced but no such target exists.


CMake Error at src/bindings/io/uv/CMakeLists.txt:61 (add_executable):
  No SOURCES given to target: testio_uv


CMake Error at src/bindings/io/glib/CMakeLists.txt:52 (add_executable):
  No SOURCES given to target: testio_glib
```


## System Information

- Elektra Version: master

## Workaround

```
-DBUILD_TESTING=OFF
-DPLUGINS="ALL;-multifile;-simpleini"
-DBINDINGS="MAINTAINED;-intercept_env;-intercept_fs;-io_uv;-io_ev;-io_glib"
```
works

--- a/doc/news/_preparation_next_release.md
+++ b/doc/news/_preparation_next_release.md
@@ -166,7 +166,7 @@ you up to date with the multi-language support provided by Elektra.
 
 ### CMake
 
-- <<TODO>>
+- Disable binding tests when `BUILD_TESTING` is disabled. _(Robert Sowula)_
 - <<TODO>>
 - <<TODO>>
 

--- a/scripts/jenkins/Jenkinsfile
+++ b/scripts/jenkins/Jenkinsfile
@@ -91,6 +91,7 @@ CMAKE_FLAGS_ASAN = ['ENABLE_ASAN': 'ON']
 CMAKE_FLAGS_DEBUG = ['ENABLE_DEBUG': 'ON']
 CMAKE_FLAGS_LOGGER = ['ENABLE_LOGGER': 'ON']
 CMAKE_FLAGS_OPTIMIZATIONS_OFF = ['ENABLE_OPTIMIZATIONS': 'OFF']
+CMAKE_FLAGS_BUILD_TESTING_OFF = ['BUILD_TESTING': 'OFF']
 
 // CMAKE_FLAGS_BUILD_SHARED = ['BUILD_SHARED': 'ON']  is ON per default
 CMAKE_FLAGS_BUILD_FULL = ['BUILD_FULL': 'ON']
@@ -629,6 +630,16 @@ def generateFullBuildStages() {
       'KDB_DEFAULT_RESOLVER': xdgResolver
     ],
     [TEST.ALL, TEST.MEM]
+  )
+
+  // Build Elektra without tests
+  tasks << buildAndTest(
+    "debian-buster-full-build-testing-off",
+    DOCKER_IMAGES.buster,
+    CMAKE_FLAGS_BUILD_ALL +
+      CMAKE_FLAGS_BUILD_TESTING_OFF
+    ,
+    []
   )
 
   // Build Elektra without optimizations

--- a/src/bindings/io/doc/CMakeLists.txt
+++ b/src/bindings/io/doc/CMakeLists.txt
@@ -33,24 +33,26 @@ install (FILES ${IO_VARIANT_HDR_FILES} DESTINATION include/${TARGET_INCLUDE_FOLD
 
 # install ( FILES "${CMAKE_CURRENT_BINARY_DIR}/${IO_VARIANT_LIBRARY}.pc" DESTINATION lib${LIB_SUFFIX}/${TARGET_PKGCONFIG_FOLDER} )
 
-# Build test
-set (testexename testio_${BINDING_VARIANT})
-set (ADDITIONAL_TEST_LIBRARIES "m")
+if (BUILD_TESTING)
+	# Build test
+	set (testexename testio_${BINDING_VARIANT})
+	set (ADDITIONAL_TEST_LIBRARIES "m")
 
-set (TEST_SOURCES $<TARGET_OBJECTS:cframework>)
-add_headers (TEST_SOURCES)
-file (GLOB IO_TEST_SRC_FILES "${CMAKE_SOURCE_DIR}/src/bindings/io/test/test*.c")
-list (APPEND TEST_SOURCES ${IO_TEST_SRC_FILES})
-list (APPEND TEST_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/testio_${BINDING_VARIANT}.c")
+	set (TEST_SOURCES $<TARGET_OBJECTS:cframework>)
+	add_headers (TEST_SOURCES)
+	file (GLOB IO_TEST_SRC_FILES "${CMAKE_SOURCE_DIR}/src/bindings/io/test/test*.c")
+	list (APPEND TEST_SOURCES ${IO_TEST_SRC_FILES})
+	list (APPEND TEST_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/testio_${BINDING_VARIANT}.c")
 
-add_executable (${testexename} ${TEST_SOURCES})
+	add_executable (${testexename} ${TEST_SOURCES})
 
-target_include_directories (${testexename} PUBLIC "${CMAKE_SOURCE_DIR}/tests/cframework")
+	target_include_directories (${testexename} PUBLIC "${CMAKE_SOURCE_DIR}/tests/cframework")
 
-target_link_elektra (${testexename} elektra-kdb elektra-plugin elektra-io ${IO_VARIANT_LIBRARY} ${ADDITIONAL_TEST_LIBRARIES})
+	target_link_elektra (${testexename} elektra-kdb elektra-plugin elektra-io ${IO_VARIANT_LIBRARY} ${ADDITIONAL_TEST_LIBRARIES})
 
-add_test (
-	NAME ${testexename}
-	COMMAND "${CMAKE_BINARY_DIR}/bin/${testexename}" "${CMAKE_CURRENT_SOURCE_DIR}"
-	WORKING_DIRECTORY "${WORKING_DIRECTORY}")
-set_property (TEST ${testexename} PROPERTY ENVIRONMENT "LD_LIBRARY_PATH=${CMAKE_BINARY_DIR}/lib")
+	add_test (
+		NAME ${testexename}
+		COMMAND "${CMAKE_BINARY_DIR}/bin/${testexename}" "${CMAKE_CURRENT_SOURCE_DIR}"
+		WORKING_DIRECTORY "${WORKING_DIRECTORY}")
+	set_property (TEST ${testexename} PROPERTY ENVIRONMENT "LD_LIBRARY_PATH=${CMAKE_BINARY_DIR}/lib")
+endif ()

--- a/src/bindings/io/ev/CMakeLists.txt
+++ b/src/bindings/io/ev/CMakeLists.txt
@@ -34,33 +34,35 @@ else () # Build library
 		DESTINATION lib${LIB_SUFFIX}/${TARGET_PKGCONFIG_FOLDER}
 		COMPONENT libelektra-dev)
 
-	# Build test
-	set (testexename testio_${BINDING_VARIANT})
+	if (BUILD_TESTING)
+		# Build test
+		set (testexename testio_${BINDING_VARIANT})
 
-	set (TEST_SOURCES $<TARGET_OBJECTS:cframework>)
-	add_headers (TEST_SOURCES)
-	file (GLOB IO_TEST_SRC_FILES "${CMAKE_SOURCE_DIR}/src/bindings/io/test/test*.c")
-	list (APPEND TEST_SOURCES ${IO_TEST_SRC_FILES})
-	list (APPEND TEST_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/testio_${BINDING_VARIANT}.c")
+		set (TEST_SOURCES $<TARGET_OBJECTS:cframework>)
+		add_headers (TEST_SOURCES)
+		file (GLOB IO_TEST_SRC_FILES "${CMAKE_SOURCE_DIR}/src/bindings/io/test/test*.c")
+		list (APPEND TEST_SOURCES ${IO_TEST_SRC_FILES})
+		list (APPEND TEST_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/testio_${BINDING_VARIANT}.c")
 
-	if (BUILD_FULL OR BUILD_STATIC) # add sources for elektra-io-ev for static and full builds
-		list (APPEND TEST_SOURCES $<TARGET_OBJECTS:${IO_VARIANT_LIBRARY}-objects>)
+		if (BUILD_FULL OR BUILD_STATIC) # add sources for elektra-io-ev for static and full builds
+			list (APPEND TEST_SOURCES $<TARGET_OBJECTS:${IO_VARIANT_LIBRARY}-objects>)
+		endif ()
+
+		add_executable (${testexename} ${TEST_SOURCES})
+
+		target_include_directories (${testexename} PUBLIC "${CMAKE_SOURCE_DIR}/tests/cframework")
+		target_include_directories (${testexename} PUBLIC SYSTEM ${LIBEV_INCLUDE_DIRS})
+
+		target_link_elektra (${testexename} elektra-kdb elektra-plugin elektra-io ${IO_VARIANT_LIBRARY} m)
+		if (BUILD_FULL OR BUILD_STATIC)
+			target_link_libraries (${testexename} ${LIBEV_LIBRARIES})
+		endif ()
+
+		add_test (
+			NAME ${testexename}
+			COMMAND "${CMAKE_BINARY_DIR}/bin/${testexename}" "${CMAKE_CURRENT_SOURCE_DIR}"
+			WORKING_DIRECTORY "${WORKING_DIRECTORY}")
+
+		add_subdirectory (example)
 	endif ()
-
-	add_executable (${testexename} ${TEST_SOURCES})
-
-	target_include_directories (${testexename} PUBLIC "${CMAKE_SOURCE_DIR}/tests/cframework")
-	target_include_directories (${testexename} PUBLIC SYSTEM ${LIBEV_INCLUDE_DIRS})
-
-	target_link_elektra (${testexename} elektra-kdb elektra-plugin elektra-io ${IO_VARIANT_LIBRARY} m)
-	if (BUILD_FULL OR BUILD_STATIC)
-		target_link_libraries (${testexename} ${LIBEV_LIBRARIES})
-	endif ()
-
-	add_test (
-		NAME ${testexename}
-		COMMAND "${CMAKE_BINARY_DIR}/bin/${testexename}" "${CMAKE_CURRENT_SOURCE_DIR}"
-		WORKING_DIRECTORY "${WORKING_DIRECTORY}")
-
-	add_subdirectory (example)
 endif ()

--- a/src/bindings/io/glib/CMakeLists.txt
+++ b/src/bindings/io/glib/CMakeLists.txt
@@ -40,33 +40,35 @@ else ()
 		COMPONENT libelektra-dev)
 
 	# Build test
-	set (testexename testio_${BINDING_VARIANT})
+	if (BUILD_TESTING)
+		set (testexename testio_${BINDING_VARIANT})
 
-	set (TEST_SOURCES $<TARGET_OBJECTS:cframework>)
-	add_headers (TEST_SOURCES)
-	file (GLOB IO_TEST_SRC_FILES "${CMAKE_SOURCE_DIR}/src/bindings/io/test/test*.c")
-	list (APPEND TEST_SOURCES ${IO_TEST_SRC_FILES})
-	list (APPEND TEST_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/testio_${BINDING_VARIANT}.c")
+		set (TEST_SOURCES $<TARGET_OBJECTS:cframework>)
+		add_headers (TEST_SOURCES)
+		file (GLOB IO_TEST_SRC_FILES "${CMAKE_SOURCE_DIR}/src/bindings/io/test/test*.c")
+		list (APPEND TEST_SOURCES ${IO_TEST_SRC_FILES})
+		list (APPEND TEST_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/testio_${BINDING_VARIANT}.c")
 
-	if (BUILD_FULL OR BUILD_STATIC) # add sources for elektra-io-glib for static and full builds
-		list (APPEND TEST_SOURCES $<TARGET_OBJECTS:${IO_VARIANT_LIBRARY}-objects>)
+		if (BUILD_FULL OR BUILD_STATIC) # add sources for elektra-io-glib for static and full builds
+			list (APPEND TEST_SOURCES $<TARGET_OBJECTS:${IO_VARIANT_LIBRARY}-objects>)
+		endif ()
+
+		add_executable (${testexename} ${TEST_SOURCES})
+
+		target_include_directories (${testexename} PUBLIC "${CMAKE_SOURCE_DIR}/tests/cframework")
+		target_include_directories (${testexename} PUBLIC SYSTEM ${GLib_INCLUDE_DIRS})
+
+		target_link_elektra (${testexename} elektra-kdb elektra-plugin elektra-io ${IO_VARIANT_LIBRARY} m)
+		if (BUILD_FULL OR BUILD_STATIC)
+			target_link_libraries (${testexename} ${GLib_LIBRARY})
+		endif ()
+
+		add_test (
+			NAME ${testexename}
+			COMMAND "${CMAKE_BINARY_DIR}/bin/${testexename}" "${CMAKE_CURRENT_SOURCE_DIR}"
+			WORKING_DIRECTORY "${WORKING_DIRECTORY}")
+		set_property (TEST ${testexename} PROPERTY ENVIRONMENT "LD_LIBRARY_PATH=${CMAKE_BINARY_DIR}/lib")
+
+		add_subdirectory (example)
 	endif ()
-
-	add_executable (${testexename} ${TEST_SOURCES})
-
-	target_include_directories (${testexename} PUBLIC "${CMAKE_SOURCE_DIR}/tests/cframework")
-	target_include_directories (${testexename} PUBLIC SYSTEM ${GLib_INCLUDE_DIRS})
-
-	target_link_elektra (${testexename} elektra-kdb elektra-plugin elektra-io ${IO_VARIANT_LIBRARY} m)
-	if (BUILD_FULL OR BUILD_STATIC)
-		target_link_libraries (${testexename} ${GLib_LIBRARY})
-	endif ()
-
-	add_test (
-		NAME ${testexename}
-		COMMAND "${CMAKE_BINARY_DIR}/bin/${testexename}" "${CMAKE_CURRENT_SOURCE_DIR}"
-		WORKING_DIRECTORY "${WORKING_DIRECTORY}")
-	set_property (TEST ${testexename} PROPERTY ENVIRONMENT "LD_LIBRARY_PATH=${CMAKE_BINARY_DIR}/lib")
-
-	add_subdirectory (example)
 endif ()

--- a/src/bindings/io/uv/CMakeLists.txt
+++ b/src/bindings/io/uv/CMakeLists.txt
@@ -58,8 +58,8 @@ else ()
 		list (APPEND TEST_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/testio_${BINDING_VARIANT}.c")
 
 		if (BUILD_FULL OR BUILD_STATIC)
-			list (APPEND TEST_SOURCES $<TARGET_OBJECTS:${IO_UV_LIBRARY}-objects>) # add sources for elektra-io-uv for static and full
-			# builds
+			# add sources for elektra-io-uv for static and full builds
+			list (APPEND TEST_SOURCES $<TARGET_OBJECTS:${IO_UV_LIBRARY}-objects>)
 		endif ()
 
 		add_executable (${TESTEXENAME} ${TEST_SOURCES})

--- a/src/bindings/io/uv/CMakeLists.txt
+++ b/src/bindings/io/uv/CMakeLists.txt
@@ -47,40 +47,43 @@ else ()
 		DESTINATION lib${LIB_SUFFIX}/${TARGET_PKGCONFIG_FOLDER}
 		COMPONENT libelektra-dev)
 
-	# Build test
-	set (TESTEXENAME testio_${BINDING_VARIANT})
+	if (BUILD_TESTING)
+		# Build test
+		set (TESTEXENAME testio_${BINDING_VARIANT})
 
-	set (TEST_SOURCES $<TARGET_OBJECTS:cframework>)
-	add_headers (TEST_SOURCES)
-	file (GLOB IO_TEST_SRC_FILES "${CMAKE_SOURCE_DIR}/src/bindings/io/test/test*.c")
-	list (APPEND TEST_SOURCES ${IO_TEST_SRC_FILES})
-	list (APPEND TEST_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/testio_${BINDING_VARIANT}.c")
+		set (TEST_SOURCES $<TARGET_OBJECTS:cframework>)
+		add_headers (TEST_SOURCES)
+		file (GLOB IO_TEST_SRC_FILES "${CMAKE_SOURCE_DIR}/src/bindings/io/test/test*.c")
+		list (APPEND TEST_SOURCES ${IO_TEST_SRC_FILES})
+		list (APPEND TEST_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/testio_${BINDING_VARIANT}.c")
 
-	if (BUILD_FULL OR BUILD_STATIC)
-		list (APPEND TEST_SOURCES $<TARGET_OBJECTS:${IO_UV_LIBRARY}-objects>) # add sources for elektra-io-uv for static and full
-		# builds
+		if (BUILD_FULL OR BUILD_STATIC)
+			list (APPEND TEST_SOURCES $<TARGET_OBJECTS:${IO_UV_LIBRARY}-objects>) # add sources for elektra-io-uv for static and full
+			# builds
+		endif ()
+
+		add_executable (${TESTEXENAME} ${TEST_SOURCES})
+
+		target_include_directories (${TESTEXENAME} PUBLIC "${CMAKE_SOURCE_DIR}/tests/cframework" "${libuv_INCLUDE_DIRS}")
+
+		target_link_elektra (${TESTEXENAME} elektra-kdb elektra-plugin elektra-io ${IO_UV_LIBRARY} m)
+		if (BUILD_FULL OR BUILD_STATIC)
+			target_link_libraries (${TESTEXENAME} "${libuv_LIBRARIES}" m)
+		endif ()
+
+		if (libuv_VERSION VERSION_LESS "1.0")
+			target_compile_definitions (${TESTEXENAME} PRIVATE "HAVE_LIBUV0")
+		else ()
+			target_compile_definitions (${TESTEXENAME} PRIVATE "HAVE_LIBUV1")
+		endif ()
+
+		add_test (
+			NAME ${TESTEXENAME}
+			COMMAND "${CMAKE_BINARY_DIR}/bin/${TESTEXENAME}" "${CMAKE_CURRENT_SOURCE_DIR}"
+			WORKING_DIRECTORY "${WORKING_DIRECTORY}")
+		set_property (TEST ${TESTEXENAME} PROPERTY ENVIRONMENT "LD_LIBRARY_PATH=${CMAKE_BINARY_DIR}/lib")
+
+		add_subdirectory (example)
+
 	endif ()
-
-	add_executable (${TESTEXENAME} ${TEST_SOURCES})
-
-	target_include_directories (${TESTEXENAME} PUBLIC "${CMAKE_SOURCE_DIR}/tests/cframework" "${libuv_INCLUDE_DIRS}")
-
-	target_link_elektra (${TESTEXENAME} elektra-kdb elektra-plugin elektra-io ${IO_UV_LIBRARY} m)
-	if (BUILD_FULL OR BUILD_STATIC)
-		target_link_libraries (${TESTEXENAME} "${libuv_LIBRARIES}" m)
-	endif ()
-
-	if (libuv_VERSION VERSION_LESS "1.0")
-		target_compile_definitions (${TESTEXENAME} PRIVATE "HAVE_LIBUV0")
-	else ()
-		target_compile_definitions (${TESTEXENAME} PRIVATE "HAVE_LIBUV1")
-	endif ()
-
-	add_test (
-		NAME ${TESTEXENAME}
-		COMMAND "${CMAKE_BINARY_DIR}/bin/${TESTEXENAME}" "${CMAKE_CURRENT_SOURCE_DIR}"
-		WORKING_DIRECTORY "${WORKING_DIRECTORY}")
-	set_property (TEST ${TESTEXENAME} PROPERTY ENVIRONMENT "LD_LIBRARY_PATH=${CMAKE_BINARY_DIR}/lib")
-
-	add_subdirectory (example)
 endif ()

--- a/src/bindings/jna/CMakeLists.txt
+++ b/src/bindings/jna/CMakeLists.txt
@@ -54,9 +54,10 @@ if (Java_JAVAC_EXECUTABLE)
 				set (GRADLE_TEST_EXCLUDES "")
 				java_test_needs_plugin (GOptsTest gopts)
 
-				if ((CMAKE_SYSTEM_NAME STREQUAL "Darwin") AND NOT (CMAKE_SYSTEM_VERSION VERSION_LESS 15))
+				if ((CMAKE_SYSTEM_NAME STREQUAL "Darwin") AND NOT (CMAKE_SYSTEM_VERSION VERSION_LESS 15) OR (NOT BUILD_TESTING))
 					# we cannot set DYLD_LIBRARY_PATH on new macOS versions, making the kdb tests fail if its not
 					# installed in the system
+					# Tests should only be built if BUILD_TESTING is enabled.
 					java_exclude_test (ExceptionMapperIT)
 					java_exclude_test (ExceptionMapperTest)
 					java_exclude_test (GOptsTest)
@@ -167,7 +168,7 @@ if (Java_JAVAC_EXECUTABLE)
 					COMPONENT java-elektra)
 
 				# add gradle test to execute with ctest in the testing phase
-				if (NOT ENABLE_ASAN)
+				if (BUILD_TESTING AND NOT ENABLE_ASAN)
 					add_test (
 						NAME testjna_gradle
 						COMMAND ${GRADLE_EXECUTABLE} -q --console=plain test

--- a/src/bindings/jna/CMakeLists.txt
+++ b/src/bindings/jna/CMakeLists.txt
@@ -54,10 +54,11 @@ if (Java_JAVAC_EXECUTABLE)
 				set (GRADLE_TEST_EXCLUDES "")
 				java_test_needs_plugin (GOptsTest gopts)
 
-				if ((CMAKE_SYSTEM_NAME STREQUAL "Darwin") AND NOT (CMAKE_SYSTEM_VERSION VERSION_LESS 15) OR (NOT BUILD_TESTING))
+				if ((CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+				    AND NOT (CMAKE_SYSTEM_VERSION VERSION_LESS 15)
+				    OR (NOT BUILD_TESTING))
 					# we cannot set DYLD_LIBRARY_PATH on new macOS versions, making the kdb tests fail if its not
-					# installed in the system
-					# Tests should only be built if BUILD_TESTING is enabled.
+					# installed in the system Tests should only be built if BUILD_TESTING is enabled.
 					java_exclude_test (ExceptionMapperIT)
 					java_exclude_test (ExceptionMapperTest)
 					java_exclude_test (GOptsTest)

--- a/src/bindings/jna/CMakeLists.txt
+++ b/src/bindings/jna/CMakeLists.txt
@@ -58,7 +58,7 @@ if (Java_JAVAC_EXECUTABLE)
 				    AND NOT (CMAKE_SYSTEM_VERSION VERSION_LESS 15)
 				    OR (NOT BUILD_TESTING))
 					# we cannot set DYLD_LIBRARY_PATH on new macOS versions, making the kdb tests fail if its not
-					# installed in the system Tests should only be built if BUILD_TESTING is enabled.
+					# installed in the system
 					java_exclude_test (ExceptionMapperIT)
 					java_exclude_test (ExceptionMapperTest)
 					java_exclude_test (GOptsTest)

--- a/src/bindings/rust/CMakeLists.txt
+++ b/src/bindings/rust/CMakeLists.txt
@@ -32,7 +32,9 @@ if (CARGO_EXECUTABLE)
 
 	add_custom_target (rust ALL DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/target/release/libelektra.rlib")
 
-	if (BUILD_TESTING AND (NOT ENABLE_ASAN) AND BUILD_SHARED)
+	if (BUILD_TESTING
+	    AND (NOT ENABLE_ASAN)
+	    AND BUILD_SHARED)
 		add_test (
 			NAME test_rust_elektra
 			COMMAND ${CARGO_EXECUTABLE} test

--- a/src/bindings/rust/CMakeLists.txt
+++ b/src/bindings/rust/CMakeLists.txt
@@ -32,7 +32,7 @@ if (CARGO_EXECUTABLE)
 
 	add_custom_target (rust ALL DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/target/release/libelektra.rlib")
 
-	if ((NOT ENABLE_ASAN) AND BUILD_SHARED)
+	if (BUILD_TESTING AND (NOT ENABLE_ASAN) AND BUILD_SHARED)
 		add_test (
 			NAME test_rust_elektra
 			COMMAND ${CARGO_EXECUTABLE} test

--- a/src/bindings/swig/lua/CMakeLists.txt
+++ b/src/bindings/swig/lua/CMakeLists.txt
@@ -49,7 +49,9 @@ else ()
 	include_directories (${CMAKE_SOURCE_DIR}/src/swig/)
 
 	if (LUA_EXECUTABLE_VERSION_MATCHED)
-		add_subdirectory (tests)
+		if (BUILD_TESTING)
+			add_subdirectory (tests)
+		endif ()
 	else ()
 		message (WARNING "Lua interpreter does not match Lua libraries version, "
 				 "needed for SWIG Lua bindings tests. Tests will not be executed.")

--- a/src/bindings/swig/python/CMakeLists.txt
+++ b/src/bindings/swig/python/CMakeLists.txt
@@ -85,8 +85,10 @@ else ()
 		POST_BUILD
 		COMMAND ${CMAKE_COMMAND} -E copy "${CMAKE_CURRENT_SOURCE_DIR}/__init__.py" ${CMAKE_SWIG_OUTDIR}/)
 
-	# add tests
-	add_subdirectory (tests)
+	if (BUILD_TESTING)
+		# add tests
+		add_subdirectory (tests)
+	endif ()
 
 	# tools module
 	#
@@ -113,8 +115,10 @@ else ()
 		set_target_properties (_swig-python-tools PROPERTIES OUTPUT_NAME _tools)
 		set_source_files_properties (${swig_generated_file_fullname} PROPERTIES COMPILE_FLAGS "${SWIG_COMPILE_FLAGS}")
 
-		# add tests
-		add_subdirectory (tests_tools)
+		if (BUILD_TESTING)
+			# add tests
+			add_subdirectory (tests_tools)
+		endif ()
 	endif (${SWIG_VERSION} VERSION_LESS "4.0.0")
 
 	# install package

